lib/client/tncon/tncon.c

```diff
@@ -0,0 +1,883 @@
+				WriteToBuffer((char *)ALT_CTRL_PF11_KEY, strlen(ALT_CTRL_PF11_KEY));
+
+			else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & RIGHT_ALT_PRESSED) ||
+				(dwControlKeyState & LEFT_ALT_PRESSED)))
+				WriteToBuffer((char *)SHIFT_ALT_PF11_KEY, strlen(SHIFT_ALT_PF11_KEY));
+
+			else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & LEFT_CTRL_PRESSED) ||
+				(dwControlKeyState & RIGHT_CTRL_PRESSED)))
+				WriteToBuffer((char *)SHIFT_CTRL_PF11_KEY, strlen(SHIFT_CTRL_PF11_KEY));
+			break;
+		case VK_F12:
+			if (dwControlKeyState == 0)
+				WriteToBuffer((char *)PF12_KEY, strlen(PF12_KEY));
+
+			else if (dwControlKeyState == SHIFT_PRESSED)
+				WriteToBuffer((char *)SHIFT_PF12_KEY, strlen(SHIFT_PF12_KEY));
+
+			else if (dwControlKeyState == LEFT_CTRL_PRESSED || dwControlKeyState == RIGHT_CTRL_PRESSED)
+				WriteToBuffer((char *)CTRL_PF12_KEY, strlen(CTRL_PF12_KEY));
+
+			else if (dwControlKeyState == LEFT_ALT_PRESSED || dwControlKeyState == RIGHT_ALT_PRESSED)
+				WriteToBuffer((char *)ALT_PF12_KEY, strlen(ALT_PF12_KEY));
+
+			else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & RIGHT_ALT_PRESSED) ||
+				(dwControlKeyState & LEFT_ALT_PRESSED)) && ((dwControlKeyState & LEFT_CTRL_PRESSED) ||
+				(dwControlKeyState & RIGHT_CTRL_PRESSED)))
+				WriteToBuffer((char *)SHIFT_ALT_CTRL_PF12_KEY, strlen(SHIFT_ALT_CTRL_PF12_KEY));
+
+			else if ((dwControlKeyState & RIGHT_ALT_PRESSED) || (dwControlKeyState & LEFT_ALT_PRESSED) &&
+				((dwControlKeyState & LEFT_CTRL_PRESSED) || (dwControlKeyState & RIGHT_CTRL_PRESSED)))
+				WriteToBuffer((char *)ALT_CTRL_PF12_KEY, strlen(ALT_CTRL_PF12_KEY));
+
+			else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & RIGHT_ALT_PRESSED) ||
+				(dwControlKeyState & LEFT_ALT_PRESSED)))
+				WriteToBuffer((char *)SHIFT_ALT_PF12_KEY, strlen(SHIFT_ALT_PF12_KEY));
+
+			else if ((dwControlKeyState & SHIFT_PRESSED) && ((dwControlKeyState & LEFT_CTRL_PRESSED) ||
+				(dwControlKeyState & RIGHT_CTRL_PRESSED)))
+				WriteToBuffer((char *)SHIFT_CTRL_PF12_KEY, strlen(SHIFT_CTRL_PF12_KEY));
+			break;
+		default:
+			if (inputRecord.Event.KeyEvent.uChar.UnicodeChar != L'\0') {
+				if ((dwControlKeyState & LEFT_ALT_PRESSED) || (dwControlKeyState & RIGHT_ALT_PRESSED)) {
+					memset(tmp_buf, 0, sizeof(tmp_buf));
+					tmp_buf[0] = '\x1b';
+					memcpy(tmp_buf + 1, (char *)octets, n);
+					WriteToBuffer(tmp_buf, n + 1);
+				}
+				else
+					WriteToBuffer((char *)octets, n);
+				break;
+			}
+		}
+	}
+}
+
+// ReadInputContinuous reads all console input events until the program exits.
+// This is a blocking call and will not return; callers should spawn this in
+// a background thread and subscribe to events via the Go interface.
+void ReadInputContinuous() {
+	HANDLE hInput = GetStdHandle(STD_INPUT_HANDLE);
+	char buf[32];
+
+	while (true) {
+		ReadConsoleForTermEmul(hInput, buf, 32);
+	}
```

Comment:
maybe you can clone stdin handle and then close it to end this function?
having it run forever creates a lot of limitations